modules/apps/portal-security-password/portal-security-password-service/src/main/java/com/liferay/portal/security/password/service/impl/HashAlgorithmEntryLocalServiceImpl.java

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.security.password.service.impl;
+
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.aop.AopService;
+import com.liferay.portal.configuration.metatype.bnd.util.ConfigurableUtil;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.service.PortalPreferencesLocalService;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.security.password.configuration.PasswordConfiguration;
+import com.liferay.portal.security.password.model.HashAlgorithmEntry;
+import com.liferay.portal.security.password.service.base.HashAlgorithmEntryLocalServiceBaseImpl;
+
+import java.util.Map;
+
+import javax.portlet.PortletPreferences;
+
+import org.json.JSONObject;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * The implementation of the hash algorithm entry local service.
+ *
+ * <p>
+ * All custom service methods should be put in this class. Whenever methods are added, rerun ServiceBuilder to copy their definitions into the <code>com.liferay.portal.security.password.service.HashAlgorithmEntryLocalService</code> interface.
+ *
+ * <p>
+ * This is a local service. Methods of this service will not have security checks based on the propagated JAAS credentials because this service can only be accessed from within the same VM.
+ * </p>
+ *
+ * @author arthurchan35
+ * @see HashAlgorithmEntryLocalServiceBaseImpl
+ */
+@Component(
+	configurationPid = "com.liferay.portal.security.password.configuration.PasswordConfiguration",
```

Comment:
hey @arthurchan35,
I have been looking around and it looks fine to have `configurationPid` in the localServiceImpl... we already have examples in other modules. 

Thx a ton!